service/src/main/java/uk/gov/hmcts/reform/fpl/service/robotics/RoboticsDataService.java

```diff
@@ -71,6 +78,13 @@ public RoboticsData prepareRoboticsData(final CaseData caseData, final Long case
             .owningCourt(toInt(hmctsCourtLookupConfiguration.getCourt(caseData.getCaseLocalAuthority()).getCourtCode()))
             .caseId(caseId)
             .build();
+
+        if (!validRoboticsData(roboticsData)) {
+            throw new RoboticsDataException(String.format("failed validation with these error(s) %s",
+                validatorService.validationErrors(roboticsData)));
+        }
```

Comment:
validation happens twice here, inside validRoboticsData and to build message. Currently is it not big deal but if validation will change and becomes time consuming (e.g external call) then it would be waste, maybe:
        
List<String> validationErrors = validatorService.validationErrors(roboticsData);

        if(isNotEmpty(validationErrors)){
            throw new RoboticsDataException(String.format("failed validation with these error(s) %s", validationErrors));
        }